File: dotNET/PFProSDK/src/PayPal/Payments/Communication/RetryState.cs
#region "Imports"

using PayPal.Payments.Common.Logging;
using PayPal.Payments.Common.Utility;

#endregion

namespace PayPal.Payments.Communication
{
	/// <summary>
	/// Represents RetryState.
	/// </summary>
	abstract internal class RetryState : PaymentState
	{
		#region "Constructors"

		/// <summary>
		/// Copy Constructor for RetryState.
		/// </summary>
		/// <param name="CurrentPmtState">Current Payment State.</param>
		public RetryState(PaymentState CurrentPmtState) : base(CurrentPmtState)
		{
		}

		#endregion

		#region "Functions"

		/// <summary>
		/// Execute function.
		/// </summary>
		public override void Execute()
		{
			Logger.Instance.Log("PayPal.Payments.Communication.RetryState.Execute(): Entered.", PayflowConstants.SEVERITY_DEBUG);
			mConnection.Disconnect();
			SetStateSuccess();
			Logger.Instance.Log("PayPal.Payments.Communication.RetryState.Execute(): Exiting.", PayflowConstants.SEVERITY_DEBUG);
		}

		#endregion
	}
}

File: java/src/sdk/base/paypal/payflow/RetryState.java
package paypal.payflow;



/**
 * Represents RetryState.
 */
abstract class RetryState extends PaymentState {

    /**
     * Copy Constructor for RetryState.
     *
     * @param currentPmtState PaymentState
     */
    public RetryState(PaymentState currentPmtState) {
        super(currentPmtState);
        Logger.getInstance().log("paypal.payflow.RetryState.RetryState(PaymentState) : Entered", PayflowConstants.SEVERITY_DEBUG);
        Logger.getInstance().log("paypal.payflow.RetryState.RetryState(PaymentState) : Exiting", PayflowConstants.SEVERITY_DEBUG);
    }

    /**
     * Execute function.
     */
    public void execute() {
        Logger.getInstance().log("paypal.payflow.RetryState.Execute(): Entered", PayflowConstants.SEVERITY_DEBUG);
        mConnection.disconnect();
        setStateSuccess();
        Logger.getInstance().log("paypal.payflow.RetryState.Execute(): Exiting", PayflowConstants.SEVERITY_DEBUG);
    }

}